### PR TITLE
CRUD actions

### PR DESCRIPTION
**Pull request by hofmannedv**
Extended the CRUD documentation my making the terms more clear that are in use.

This is part one. More changes are on its way.

### CODE REVIEW

**Comment by hofmannedv** [comment]
Added changes part 2.

File: chapters/crud.md [hunk 1]
```diff
@@ -4,15 +4,40 @@
...
 
-CRUD stands for Create, Read, Update, Delete. Together, they form the basis of any database engine. While Jina itself is *not* a database engine, it does provide these methods of interacting with the data stored in its indexes.
+#### Overview
 
-Before `1.0`, Jina only supported indexing (creating) and querying (reading) Documents. To update or delete a Document, you had to edit your dataset, and then rebuild the Flow and indexes. Needless to say, this could create problems if you had large datasets.
+CRUD abbreviates the four actions **C**reate, **R**ead, **U**pdate, and **D**elete. All together they form the basis of a database management system (DBMS). While Jina itself is *not* a DBMS, it does provide these methods of interacting with the data stored in its indexes. Jina uses the following terms for this:
 
-With the release of version `1.0` we are introducing **update** and **delete** operations. These are implemented across our Executors and Drivers, and allow you to update and delete Documents by their ids. 
+CRUD Term    Jina Term 
+----------   ----------
```
Comment (rutujasurve94) [inline chapters/crud.md]: Follow GitHub markdown for table creation like:

| `CRUD Term`  | `Jina Term` |
| ------------- | ------------- |
| `Create`  | Index |

File: chapters/crud.md [hunk 2]
```diff
@@ -4,15 +4,40 @@
...
+
+#### Availability in Jina
+
+Before version `1.0`, Jina only supported indexing (creating) and querying (reading) Documents. In order to update or delete a Document, you had to edit your dataset first, and then rebuild both the Flow and the corresponding indexes. Needless to say, this could create problems if you had large datasets.
+
+With the release of version `1.0` we are introducing **update** and **delete** operations. These are implemented across our Executors and Drivers, and allow you to update and delete Documents by their ids. With this, the full CRUD functionality is established and available for you in Jina.
+
+#### CRUD actions in detail
+
+What do the four CRUD actions mean, and do in Jina in particular:
```
Comment (rutujasurve94) [inline chapters/crud.md:28]: Reframe as:
#### `CRUD actions in Jina`
Remove line `What do the four CRUD actions mean, and do in Jina in particular:`

File: chapters/crud.md [hunk 3]
```diff
@@ -4,15 +4,40 @@
...
+- Query: retrieve indexed Documents based on given search criteria
+
+- Update: update an existing Document by re-indexing it
+
+- Delete: remove an existing Document from the search index
 
 ## Before you start 
 
-Study the basics [here](https://docs.jina.ai/chapters/core/introduction/index.html)
+Before you start indexing documents with Jina and searching through the database, we recommend you study the basics as explained [here](https://docs.jina.ai/chapters/core/introduction/index.html).
```
Comment (rutujasurve94) [inline chapters/crud.md]: add to: recommend you "to" study

File: chapters/crud.md [hunk 4]
```diff
@@ -4,15 +4,40 @@
...
 
-With the release of version `1.0` we are introducing **update** and **delete** operations. These are implemented across our Executors and Drivers, and allow you to update and delete Documents by their ids. 
+CRUD Term    Jina Term 
+----------   ----------
+Create       Index
+Read         Query
+Update       Update
+Delete       Delete
+
+We hope that the table above will help to avoid any possible confusion regarding the terms that are in use. We discuss the four actions in more detail below.
```
Comment (rutujasurve94) [inline chapters/crud.md:18]: Remove this line

File: chapters/crud.md (hunk 3 above)
Comment (hofmannedv) [reply chapters/crud.md]: Oh, thanks. I have added this.

File: chapters/crud.md (hunk 2 above)
Comment (hofmannedv) [reply chapters/crud.md:28]: Ok, I have changed the title of the section.
The second line ("What do the ...") was added for clarity, and we can take it out if the CRUD actions are clear for everyone.

File: chapters/crud.md (hunk 1 above)
Comment (hofmannedv) [reply chapters/crud.md]: I applied the GitHub markdown for tables.

**Comment by hofmannedv** [comment]
I applied the changes as requested by @rutujasurve94

**Comment by hofmannedv** [comment]
I commented the example code, corrected grammar, and (hopefully) fixed the numbering of the Limitations section

**Review by FionnD — APPROVED** [review]

**Pull request merged**